### PR TITLE
Balanced 1.1

### PR DESCRIPTION
**Pull request by rserna2010**

### CODE REVIEW

File: rentmybike/models/users.py [hunk 1]
```diff
@@ -50,16 +50,16 @@ def is_authenticated(self):
     @property
     def balanced_account(self):
```
Comment (mjallday) [inline rentmybike/models/users.py]: `balanced_account` -> `balanced_customer`

File: rentmybike/models/users.py [hunk 2]
```diff
@@ -79,40 +79,26 @@ def create_balanced_account(self, card_uri=None,
...
-            else:
                 raise
         return account
 
     def _create_balanced_merchant(self, merchant_data):
         marketplace = balanced.Marketplace.my_marketplace
         try:
-            account = marketplace.create_merchant(self.email_address,
-                name=self.name, merchant=merchant_data)
+            account = balanced.Customer(email=self.email,
```
Comment (mjallday) [inline rentmybike/models/users.py]: this isn't how it's done anymore. you do not explicitly create a merchant, you just pass in the merchant data on the customer object e.g. 

```
customer.name = 'foo'
customer.tax_id = '4445'
customer.dob_month = 5
```

File: rentmybike/models/users.py [hunk 3]
```diff
@@ -152,14 +137,15 @@ def add_card(self, card_uri):
...
-            account = balanced.Marketplace.create_buyer(
-                email_address=self.email_address, card_uri=card_uri,
-                name=self.name)
+            account = balanced.Customer(
+                email=self.email, card_uri=card_uri,
+                name=self.name).save()
         else:
-            account.add_card(card_uri)
+            card = balanced.Card.fetch(card_uri)
+            card.associate_to_customer(account.href)
```
Comment (mjallday) [inline rentmybike/models/users.py:146]: you do not need to fetch the card here. you can do `customer.card_href = card_uri; customer.save()` **i think**

File: requirements.txt [hunk 4]
```diff
@@ -8,7 +8,7 @@ MarkupSafe==0.15
 SQLAlchemy==0.7.6
 WTForms==1.0.1
 Werkzeug==0.8.3
-balanced>=0.8.6
+balanced>=1.0.0
```
Comment (mjallday) [inline requirements.txt]: `balanced>=1.1,<1.2`

File: tests/__init__.py [hunk 5]
```diff
@@ -28,7 +28,7 @@ class ControllerTestCase(TestCase):
     @classmethod
     def setUpClass(cls):
         super(ControllerTestCase, cls).setUpClass()
-        balanced.configure(cls.config['BALANCED_SECRET'])
+        balanced.configure('17ca08eca7a211e2ac83026ba7cac9da')
```
Comment (mjallday) [inline tests/__init__.py]: why hardcoded instead of from config?

File: tests/__init__.py [hunk 6]
```diff
@@ -59,40 +59,42 @@ def get_csrf_token(self):
...
         user_payload = {
             '_csrf_token': self.get_csrf_token(),
-            'account-email_address': email_address,
+            'account-email': email,
             'account-name': 'Bob Geldof',
             'account-password': 'ab',
-            }
+        }
         resp = self.client.post('/accounts/new', data=user_payload)
-        self.assertEqual(resp.status_code, 302)
```
Comment (mjallday) [inline tests/__init__.py:76]: what does this change mean?

File: rentmybike/settings/default.py [hunk 7]
```diff
@@ -27,7 +27,7 @@
 
 DOMAIN_URI = 'http://www.example.com'
 
-BALANCED_SECRET = '6289680a886c11e2a44a026ba7cd33d0'
+BALANCED_SECRET = '17ca08eca7a211e2ac83026ba7cac9da'
```
Comment (rserna2010) [inline rentmybike/settings/default.py]: edit back

**Pull request merged**